src/QsCompiler/SyntaxProcessor/ExpressionVerification.fs

```diff
@@ -246,11 +270,16 @@ let private VerifyConcatenation parent addError (lhsType : ResolvedType, lhsRang
 /// If a common base type exists, verifies that this base type supports equality comparison, 
 /// adding the corresponding error otherwise.
 /// If one of the given types is a missing type, also adds the corresponding ExpressionOfUnknownType error(s).
-let private VerifyEqualityComparison parent addError (lhsType : ResolvedType, lhsRange) (rhsType : ResolvedType, rhsRange) =
-    // NOTE: this may not be the behavior that we want (right now it does not matter, since we don't support equality comparison for any derived type)
-    let baseType = CommonBaseType addError (ErrorCode.ArgumentMismatchInBinaryOp, [lhsType |> toString; rhsType |> toString]) parent (lhsType, lhsRange) (rhsType, rhsRange)
-    let expected (t : ResolvedType) = t.supportsEqualityComparison
-    VerifyIsOneOf expected (ErrorCode.InvalidTypeInEqualityComparison, [baseType |> toString]) addError (baseType, rhsRange) |> ignore
+let private VerifyEqualityComparison context addError (lhsType, lhsRange) (rhsType, rhsRange) =
+    // NOTE: this may not be the behavior that we want (right now it does not matter, since we don't support equality
+    // comparison for any derived type).
+    let argumentError = ErrorCode.ArgumentMismatchInBinaryOp, [toString lhsType; toString rhsType]
+    let baseType = CommonBaseType addError argumentError context.Symbols.Parent (lhsType, lhsRange) (rhsType, rhsRange)
+    match baseType.Resolution with
+    | Result when context.Capabilities = RuntimeCapabilities.QPRGen0 ->
+        addError (ErrorCode.UnsupportedResultComparison, [context.Capabilities.ToString ()]) rhsRange
+    | BigInt | Bool | Double | Int | Pauli | Qubit | Result | String -> ()
```

Comment:
This does not properly treat invalid/missing types. I'd rather go with this: 
```suggestion
    | _ -> 
        let expected (t : ResolvedType) = t.supportsEqualityComparison
        VerifyIsOneOf expected (ErrorCode.InvalidTypeInEqualityComparison, [baseType |> toString]) addError (baseType, rhsRange) |> ignore
```

Reply:
Thanks for catching this. Do you know where I can add a test for this? Are there existing tests for invalid/missing type handling?

Reply:
You could add it anywhere to the LocalVerification tests. These tests will check the error message generated for different scenarios, and would thus easily catch e.g. that when something is already invalid, no additional error will be generated by this one.

Reply:
I added tests for this in #465.